src/libostree/ostree-repo-checkout.c

```diff
@@ -992,18 +1055,28 @@ checkout_tree_at (OstreeRepo                        *self,
                   GError                           **error)
 {
   g_auto(CheckoutState) state = { 0, };
-  // If SELinux labeling is enabled, we need to keep track of the full path string
+
+  if (options->filter)
+    state.path_buf = new_canonicalized_path_buf (options->subpath);
```

Comment:
We can switch this to `/` so that we call back with the subpath prefixed removed already. Hmm, yeah that feels more natural actually.